instrumentation/graphql/test/instrumentation/graphql/tracers/graphql_tracer_test.rb

```diff
@@ -154,6 +154,21 @@
         _(span).wont_be_nil
       end
     end
+
+    it 'traces validate with events' do
+      SomeGraphQLAppSchema.execute(
+        <<-GRAPHQL
+          {
+            nonExistentField
+          }
+        GRAPHQL
+      )
+      span = spans.find { |s| s.name == 'graphql.validate' }
+      event = span.events.find { |e| e.name == 'graphql.validation.error' }
+      _(event.attributes['message']).must_equal(
+        ["{\"message\":\"Field 'nonExistentField' doesn't exist on type 'Query'\",\"locations\":[{\"line\":2,\"column\":13}],\"path\":[\"query\",\"nonExistentField\"],\"extensions\":{\"code\":\"undefinedField\",\"typeName\":\"Query\",\"fieldName\":\"nonExistentField\"}}"]
```

Comment:
I am interested in hearing from spec SIG, vendors, and other SDK implementors about using escaped JSON text in event attribute values. It seems like there will be a bit of post processing required here to represent the event attributes.

I see e.g. in the spec when using arrays or dictionaries they recommend breaking out key-value pairs into jQuery like `application/x-www-form-urlencoded`:

https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/common.md#attributes

So maybe the attributes would be something like:

```
errors[0].message =  "Field 'nonExistentField' doesn't exist on type 'Query'"
errors[0].locations[0][line] = 2
# ...
``` 

Thoughts?

Reply:
The [JS instrumentation](https://github.com/open-telemetry/opentelemetry-js-contrib/blob/e8c207b40faa4e2de1bac95d4b9c83f0833ee455/plugins/node/opentelemetry-instrumentation-graphql/src/instrumentation.ts#L381) uses JSON string. If we supply a string with json list, jaeger at the very least pretty prints it.
![image](https://user-images.githubusercontent.com/38668303/128747638-9aafe52e-89c1-4f94-9acd-0d226875f3c2.png)

Reply:
```diff
commit 388d208c67e0deb6c2063934ec02450a9992ec63 (HEAD -> add-errors-events-to-graphql-instrumentation, origin/add-errors-events-to-graphql-instrumentation)
Author: Timur Borkhodoev <tim.borkhodoev@shopify.com>
Date:   Mon Aug 9 13:24:25 2021 -0400

    wip: better

────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
modified: instrumentation/graphql/lib/opentelemetry/instrumentation/graphql/tracers/graphql_tracer.rb
────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
@ instrumentation/graphql/lib/opentelemetry/instrumentation/graphql/tracers/graphql_tracer.rb:32 @ module OpenTelemetry

            tracer.in_span(platform_key, attributes: attributes_for(key, data)) do |span|
              yield.tap do |response|
                errors = response[:errors]&.compact&.map { |e| e.to_h.to_json } if key == 'validate'
                errors = response[:errors]&.compact&.map { |e| e.to_h }&.to_json if key == 'validate'
                unless errors.nil?
                  span.add_event(
                    'graphql.validation.error',
```